components/CategoryFilter.js

```diff
@@ -0,0 +1,26 @@
+import useSWR from "swr";
+
+export default function CategoryFilter({ onSelectCategory }) {
+  const { data: categories, isLoading, error } = useSWR("api/categories");
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+  if (error) {
+    return <p>We could not load the categories...</p>;
+  }
+
+  return (
+    <div>
+      {categories.map((category) => (
+        <button
+          key={category._id}
+          onClick={() => onSelectCategory(category.category)}
+        >
```

Comment:
```suggestion
        <button
          type="button"
          key={category._id}
          onClick={() => onSelectCategory(category.category)}
        >
```

Buttons should have `type="button"` unless you want them to submit a form